Sources/Moya/Task.swift

```diff
@@ -3,39 +3,91 @@ import Foundation
 /// Represents an HTTP task.
 public enum Task {
 
-    /// A request with no additional data.
-    case requestPlain
+    public typealias TaskParameters = [(ParameterEncoder, Encodable)]
 
-    /// A requests body set with data.
-    case requestData(Data)
+    /// A task to request some data
+    case request(rawBody: Data?, params: TaskParameters?)
```

Comment:
hmm not sure if I get this - you can specify both `rawBody` (which will be a `httpBody`) and also add params which can also append to body? 

also keep in mind that `URLEncoding` or the new encoder can also append to body

Reply:
Yes that was a flaw of this design. The idea here was that if some other parameter was to also be added as a body, it would override the rawData.
But in the 2nd version this behavior is now prevented by using only 1 kind of body param.